src/utilities/command-definition-utils.ts

```diff
@@ -0,0 +1,95 @@
+import { CommandDefinition } from "../interfaces/command-definition";
+import { CommandDefinitions } from "../modules/command-definitions";
+
+// -----------------------------------------------------------------------------------------
+// #region Public Functions
+// -----------------------------------------------------------------------------------------
+
+/**
+ * Utility functions related to working with `CommandDefinition` objects
+ */
+const CommandDefinitionUtils = {
```

Comment:
Having the `map` param be optional and default the `CommandDefinitions` throughout this file will make life easier for many implementers of utils. One way to program this to be more testable, well more "obviously" testable, would be to have a 'defaultMap' property of sorts on the utils. As opposed to having each function grab at the module. The default would be assigned to `CommandDefinitions` but folks could easily set it for testing and whatever other scenario we aren't thinking of

Reply:
Definitely 👍 I wasn't quite sure how to approach it - for flexibility and testing, it made sense to pass in each time. But I agree, the majority of the time it will be used with the `CommandDefinitions` module that we provide. I had kicked around the idea of making this a "collection" record or class of sorts, but maintaining the underlying fields and tacking on methods on felt a little weird from a typing perspective.

I'll take another stab at allowing a default map to be configured, thanks for the idea! 🚀

Reply:
Looking into this more, and I think it makes sense to take the same approach with `CommandUtils` - if no `program.Command` is passed in, it will use a default one (the global `program` instance). Should make the public API much easier to use while allowing flexibility.